packages/venia-ui/lib/queries/mergeCarts.gql.js

```diff
@@ -0,0 +1,39 @@
+import gql from 'graphql-tag';
+
+import { CartPageFragment } from '../components/CartPage/cartPageFragments.gql';
+import { CartTriggerFragment } from '../components/Header/cartTriggerFragments.gql';
+import { CheckoutPageFragment } from '../components/CheckoutPage/checkoutPageFragments.gql';
```

Comment:
It's not great that this is needed; I almost typed that all queries mounted would re-fetch on the cartId change, but they wouldn't re-fetch on merge 😞  it would almost be preferable to wait to update cartId state until they've been merged, but that sounds very imbedded into those redux actions. Oof, very fragile, but can't think of an alternative until redux is out of the picture.

The only other quick solution I can think of is a page refresh, but this sounds preferable over that for now, and we can re-visit later.